### PR TITLE
[DRAFT]For Test and Review: React component work all merged in single PR.

### PR DESCRIPTION
**Pull request by jitu5**
## Description

This is draft PR created just for test and review to see all the changes related to React component work together in one PR. 

Exposing internal state: https://github.com/kedro-org/kedro-viz/pull/1969
metadataPanel prop: https://github.com/kedro-org/kedro-viz/pull/1965
MiniMapBtn prop: https://github.com/kedro-org/kedro-viz/pull/1982
miniMap & pipelineBtn prop: https://github.com/kedro-org/kedro-viz/pull/1983
Documentation: https://github.com/kedro-org/kedro-viz/pull/1954

All the above branches are merged into this and once all the above is approved, I will merge this into `main`. 

## QA notes

You can test new options prop in `src/components/container.js` as `<App /> `is the entry point or top level component for a standalone use case.

I created 3 state `tags`, `nodeTypes` and `theme` to test when it changes on `setTimeout` after 10 secs.
Pass options prop to component to test all possible cases.
```

import React, { useEffect } from 'react';
import App from '../app';
import getPipelineData from '../../utils/data-source';
import './container.scss';

/**
 * Top-level component for the use-case where Kedro-Viz is run as a standalone
 * app rather than imported as a library/package into a larger application.
 */
const Container = () => {
  const [tags, setTags] = React.useState({
    enabled: { companies: true, shuttles: true },
  });
  const [nodeTypes, setNodeTypes] = React.useState({
    disabled: {
      parameters: false,
      task: false,
      data: false,
    },
  });
  const [theme, setTheme] = React.useState('dark');

  useEffect(() => {
    setTimeout(() => {
      setTags({
        enabled: { evaluate: true, companies: false },
      });
      setNodeTypes({
        disabled: {
          parameters: true,
          task: true,
          data: false,
        },
      });
      setTheme('light');
    }, 10000);
  }, [theme, tags, nodeTypes, expAll]);

  return (
    <>
      <App
        data={getPipelineData()}
        options={{
          expandAllPipelines: expAll,
          display: {
            globalNavigation: true,
            sidebar: true,
            miniMap: true,
            expandPipelinesBtn: true,
            exportBtn: true,
            labelBtn: true,
            layerBtn: true,
            zoomToolbar: true,
            metadataPanel: true,
          },
          tag: tags,
          nodeType: nodeTypes,
          theme: theme,
        }}
      />
    </>
  );
};
```

export default Container;

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/actions/pipelines.js
Comment (ravi-kumar-pilla) [inline src/actions/pipelines.js:138]: I saw you already addressed this in the child branch. Please look for this before we merge this to main

Comment (jitu5) [reply src/actions/pipelines.js:138]: I will not merge this branch to main but individual branch, this is just to see all code together. Let me merge above changes to current branch.

File: src/store/initial-state.js
Comment (ravi-kumar-pilla) [inline src/store/initial-state.js:39]: [Nit] Can be ignored, would like to see this as false/true in future instead of null. Or if there is any reason for this to be null, please let me know. Thank you

File: src/actions/pipelines.js
Comment (ravi-kumar-pilla) [reply src/actions/pipelines.js:138]: Just a suggestion, It would be better to merge the combined branch to main instead of individual branches. Like having child branches merge to this and then this branch to main. That way we can be sure our tests are passing in this branch and will also pass in main.

**Review by ravi-kumar-pilla — APPROVED** [review]
Apart from the [Nit] comment [here](https://github.com/kedro-org/kedro-viz/pull/1954#discussion_r1685547406), everything works as expected in local. Great work @jitu5 💯 . Thank you

**Pull request merged**
